From 68e7ee382b0f5b8aba24726abd274684e2b598ae Mon Sep 17 00:00:00 2001
From: Elizabeth Figura <zfigura@codeweavers.com>
Date: Mon, 7 Jul 2025 15:16:51 -0500
Subject: [PATCH 38/48] kernelbase: Open the reparse point in
 SetFileAttributes().

(cherry picked from commit 47cd0f7fcf496cd0552625221a6a8089099b1693)

CW-Bug-Id: #26492
---
 dlls/kernelbase/file.c  | 3 ++-
 dlls/ntdll/tests/file.c | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/dlls/kernelbase/file.c b/dlls/kernelbase/file.c
index 9ab07d78125..c91224146af 100644
--- a/dlls/kernelbase/file.c
+++ b/dlls/kernelbase/file.c
@@ -2862,7 +2862,8 @@ BOOL WINAPI DECLSPEC_HOTPATCH SetFileAttributesW( LPCWSTR name, DWORD attributes
     }
 
     InitializeObjectAttributes( &attr, &nt_name, OBJ_CASE_INSENSITIVE, 0, NULL );
-    status = NtOpenFile( &handle, SYNCHRONIZE, &attr, &io, 0, FILE_SYNCHRONOUS_IO_NONALERT );
+    status = NtOpenFile( &handle, SYNCHRONIZE, &attr, &io, 0,
+                         FILE_SYNCHRONOUS_IO_NONALERT | FILE_OPEN_REPARSE_POINT );
     RtlFreeUnicodeString( &nt_name );
 
     if (status == STATUS_SUCCESS)
diff --git a/dlls/ntdll/tests/file.c b/dlls/ntdll/tests/file.c
index 3b76fea646a..636706188fe 100644
--- a/dlls/ntdll/tests/file.c
+++ b/dlls/ntdll/tests/file.c
@@ -7034,7 +7034,7 @@ static void test_reparse_points(void)
         "got attributes %#x\n", ret );
 
     ret = GetFileAttributesW( path2 );
-    todo_wine ok( ret == FILE_ATTRIBUTE_ARCHIVE, "got attributes %#x\n", ret );
+    ok( ret == FILE_ATTRIBUTE_ARCHIVE, "got attributes %#x\n", ret );
 
     status = NtQueryAttributesFile( &attr, &basic_info );
     todo_wine ok( !status, "got %#lx\n", status );
-- 
2.53.0

